ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -795,7 +804,13 @@ public void alterTable(String catName, String dbName, String tblName, Table newT
       // Take a table snapshot and set it to newTbl.
       AcidUtils.TableSnapshot tableSnapshot = null;
       if (transactional) {
-        if (replWriteId > 0) {
+        if (AcidUtils.isTransactionalTable(newTbl) && !inReplication(newTbl) && replWriteId > 0) {
```

Comment:
Why do we not do this stuff for replication? Is replication and Cached metastore will not be supported in a single deployment?

Reply:
Initially replication will not be supported. Also, we will be using CatalogD as the cache, so I am not bringing relevant changes in CachedStore.

Reply:
I still would suggest to talk to the replication guys. There might be some surprises there.
CC: @anishek